packages/roosterjs-editor-core/lib/coreApi/getPendableFormatState.ts

```diff
@@ -0,0 +1,62 @@
+import { contains, getTagOfNode, PendableFormatNames, Position } from 'roosterjs-editor-dom';
+import { EditorCore, NodePosition, NodeType, PendableFormatState } from 'roosterjs-editor-types';
+
+/**
+ *
+ * @param core The EditorCore object
+ * @param tryGetFromCache this function will try to  get the pendable form from cached.
+ * @returns The cached format state if it exists. If the cached postion do not exist, search for pendable elements in the DOM tree and return the pendable format state.
+ */
+export function getPendableFormatState(
+    core: EditorCore,
+    tryGetFromCache: boolean = true
+): PendableFormatState {
+    const range = core.api.getSelectionRange(core, true);
+    const cachedPendableFormatState = core.pendingFormatState.pendableFormatState;
+    const cachedPosition =
+        cachedPendableFormatState && core.pendingFormatState.pendableFormatPosition.normalize();
+    const currentPosition = range && Position.getStart(range).normalize();
+    const isSamePosition = range && currentPosition.equalTo(cachedPosition);
+    if (
+        range &&
+        cachedPendableFormatState &&
+        range.collapsed &&
+        isSamePosition &&
+        tryGetFromCache
+    ) {
+        return cachedPendableFormatState;
+    } else {
+        return queryCommandState(core, currentPosition);
+    }
+}
+
+const StyleCheckers: Record<
+    PendableFormatNames,
+    (tagName: string, style: CSSStyleDeclaration) => boolean
+> = {
+    isBold: (tag, style) => tag == 'B' || tag == 'STRONG' || style.fontWeight >= '700',
```

Comment:
If I specify font-weight to "bold", what will it return here?

Reply:
It is considering  font-weight normal as bold